src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -114,11 +114,31 @@ public static ParameterizedTypeName get(ClassName rawType, TypeName... typeArgum
     return new ParameterizedTypeName(null, rawType, Arrays.asList(typeArguments));
   }
 
+  /** Returns a parameterized type, applying {@code typeArguments} to {@code rawType}. */
+  public static ParameterizedTypeName get(
+      ClassName rawType, Iterable<? extends TypeName> typeArguments) {
+    ArrayList<TypeName> typeArgumentsList = new ArrayList<>();
+    for (TypeName typeArgument : typeArguments) {
+      typeArgumentsList.add(typeArgument);
+    }
+    return new ParameterizedTypeName(null, rawType, typeArgumentsList);
+  }
+
   /** Returns a parameterized type, applying {@code typeArguments} to {@code rawType}. */
   public static ParameterizedTypeName get(Class<?> rawType, Type... typeArguments) {
     return new ParameterizedTypeName(null, ClassName.get(rawType), list(typeArguments));
   }
 
+  /** Returns a parameterized type, applying {@code typeArguments} to {@code rawType}. */
+  public static ParameterizedTypeName get(
+      Class<?> rawType, Iterable<? extends Type> typeArguments) {
+    ArrayList<Type> typeArgumentsList = new ArrayList<>();
+    for (Type typeArgument : typeArguments) {
+      typeArgumentsList.add(typeArgument);
+    }
+    return get(rawType, typeArgumentsList.toArray(new Type[0]));
```

Comment:
Doesn't https://github.com/square/javapoet/pull/788#discussion_r435436043 apply here too? Seems like the same code.

Reply:
Ah, NVM -- I see the translation that happens in `list()`.

Reply:
Yeah fixing that quickly spirals outward into more of the codebase. I'm happy to pursue such a change, but we can do so separately. Plus, this isn't particularly performance sensitive code so I figured it wasn't worth the effort.